rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java

```diff
@@ -1059,6 +1029,10 @@ private Federation deserializeFederationAccordingToVersion(
         return BridgeSerializationUtils.deserializeFederation(data, networkParameters);
     }
 
+    private <T> T safeGetFromRepository(BridgeStorageIndexKeys keyAddress, RepositoryDeserializer<T> deserializer) {
```

Comment:
Not sure it's worth adding all these new methods. We could keep using the old ones by calling `getKey()` method. At least we avoid having these methods duplicated

Reply:
But this is not duplicating these methods, it is overloading them, and in effect also avoiding having to modify all the places I touched to add a `getKey()`. The purpose of doing it in this way is to force the caller to use the enum which makes the code self-explanatory on which key is being used, what I would like will be to keep only the new method added using the enum. I can add this if we agree to have always an enum key for any entry we want to add to the storage. What do you think?

Reply:
That makes sense, let's keep only the one that receives the enum then. Thanks

Reply:
```suggestion
 private <T> T getFromRepository(BridgeStorageIndexKeys keyAddress, RepositoryDeserializer<T> deserializer) throws IOException {
        return getFromRepository(keyAddress.getKey(), deserializer);
    }

    private <T> T getFromRepository(DataWord keyAddress, RepositoryDeserializer<T> deserializer) throws IOException {
        byte[] data = repository.getStorageBytes(contractAddress, keyAddress);
        return deserializer.deserialize(data);
    }
```

After doing some analysis on how to get rid of the second method, I came to the conclusion that it may not be possible to do it. The second method it's only being used by the compound keys and so far I see no other way to it different on how is being done today.

```suggestion

    private DataWord getStorageKeyForBtcTxHashAlreadyProcessed(Sha256Hash btcTxHash) {
        return BTC_TX_HASH_AP.getCompoundKeyDataWord("-", btcTxHash.toString());
    }

    private DataWord getStorageKeyForCoinbaseInformation(Sha256Hash btcTxHash) {
        return COINBASE_INFORMATION.getCompoundKeyDataWord("-", btcTxHash.toString());
    }

    private DataWord getStorageKeyForBtcBlockIndex(Integer height) {
        return BTC_BLOCK_HEIGHT.getCompoundKeyDataWord("-", height.toString());
    }
```

```
Optional<Long> height = getFromRepository(getStorageKeyForBtcTxHashAlreadyProcessed(btcTxHash), BridgeSerializationUtils::deserializeOptionalLong);
        if (!height.isPresent()) {
            return height;
        }
```

Reply:
What do you think of this?
```
public BridgeStorageIndexKeys getCompoundKey(String delimiter, String identifier) {
        return new BridgeStorageIndexKeys(key + delimiter + identifier);
    }
```

Would that make sense? Replacing the existing `getCompoundKeyDataWord` method fro the one above? What do you think?

Also, we might want to consider renaming the class to `BridgeStorageIndexKey` (singular) since it's going to represent a single key. What do you think?

Reply:
In conclusion, after having a call and discussing different approaches, we decided to leave it as it is now, and leave this possible refactor for later due to the complexity for compound keys.